### PR TITLE
Parse book author and order date from Packt website

### PR DESCRIPTION
**Pull request by mjenczmyk**
I've extracted parsing single book data to separate method and added author and order date to parsed info. Thar will allow to sort by order date and download all books ordered in some date range or n last grabbed books.

Please, do not "merge manually", use GitHub to merge it after successful code review.

### CODE REVIEW

File: src/packtPublishingFreeEbook.py [hunk 1]
```diff
@@ -252,22 +290,18 @@ def download_books(self, titles=None, formats=None, into_folder=False):
...
-                if form in list(temp_book_data[i]['download_urls'].keys()):
-                    if form == 'code':
-                        file_type = 'zip'
-                    else:
-                        file_type = form
+                if form in list(book['download_urls']):
+                    file_type = 'zip' if form == 'code' else form
                     title = PacktAccountDataModel.convert_book_title_to_valid_string(
-                        temp_book_data[i]['title'])  # format valid pathname
+                        book['title'])  # format valid pathname
```
Comment (mjenczmyk) [inline src/packtPublishingFreeEbook.py]: No need to select by index when we have direct access to the object.

File: src/packtPublishingFreeEbook.py [hunk 2]
```diff
@@ -252,22 +290,18 @@ def download_books(self, titles=None, formats=None, into_folder=False):
         is_interactive = sys.stdout.isatty()
         for i, book in enumerate(temp_book_data):
             for form in formats:
-                if form in list(temp_book_data[i]['download_urls'].keys()):
-                    if form == 'code':
-                        file_type = 'zip'
-                    else:
-                        file_type = form
+                if form in list(book['download_urls']):
```
Comment (mjenczmyk) [inline src/packtPublishingFreeEbook.py]: Casting `dict` to list is equivalent to casting its keys to list.

File: src/packtPublishingFreeEbook.py [hunk 3]
```diff
@@ -170,7 +171,7 @@ def get_ebook_infodata(self, r):
         self._write_ebook_infodata(result_data)
         return result_data
 
-    def grab_ebook(self, log_ebook_infodata = False):
+    def grab_ebook(self, log_ebook_infodata=False):
```
Comment (mjenczmyk) [inline src/packtPublishingFreeEbook.py]: Pylama issue.

File: src/packtPublishingFreeEbook.py [hunk 4]
```diff
@@ -5,6 +5,7 @@
 import datetime as dt
 import os
 import re
+import sys
```
Comment (mjenczmyk) [inline src/packtPublishingFreeEbook.py]: Missing import.

**Comment by igbt6** [comment]
Rejected by the same reason as at #59, #54

**Comment by mjenczmyk** [comment]
Extracting order date could allow to add command to download books monthly without need to download all books anyway (which was the reason of making this pull request). Is that reason convincing enough to reopen the pull request with perspective to add such command?

**Pull request closed without merging**